fenwick_tree/cpp/fenwick_tree.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+#include <vector>
+using namespace std;
+
+#define SIZE 100000
+
+struct FenwickTree {
+    vector<int> v;
+    int len;
+
+    FenwickTree(int len = SIZE) : len(len) {
+        v.assign(len + 1, 0);
+    }
+
+    void update(int, int);
+    int query(int, int);
+    int query(int);
+};
+
+void FenwickTree::update(int i, int val) {
+    while(i <= len) {
+        v[i] += val;
+        i += i & (-i);
+    }
+}
+
+int FenwickTree::query(int l, int r) {
+    return query(r) - query(l-1);
+}
+
+int FenwickTree::query(int i) {
+    int ret = 0;
+    while(i > 0) {
+        ret += v[i];
```

Comment:
Missing space before ( in while(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*